### PR TITLE
DOCS-#4082: Add `pdf`/`epub`/`htmlzip` formats for doc builds

### LINKED ISSUE #4082: `pdf` and `epub` doc builds are missed after adding `.readthedocs.yaml`
### Describe the problem

After adding `.readthedocs.yaml` config (#4073) web-settings are ignored (https://docs.readthedocs.io/en/stable/config-file/v2.html#supported-settings). Previous releases contains `pdf/epub` doc build. Need to add those target in the new config.

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

PR adds missed after #4073 formats of doc builds 

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4082  
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by RehanSD — CHANGES_REQUESTED** [review]
Do we need to specify `htmlzip` in addition to `epub` and `pdf`? We have all three versions available for the 0.11.* and 0.12.* releases. In that case, I'd recommend just changing it to `- all`!

**Comment by prutskov** [comment]
> Do we need to specify `htmlzip` in addition to `epub` and `pdf`? We have all three versions available for the 0.11.* and 0.12.* releases. In that case, I'd recommend just changing it to `- all`!

Nice catch! `all` formats is used now

**Review by RehanSD — COMMENTED** [review]
LGTM! Could you please add this commit + your username to the 0.14.0 release notes, as per #4078? Once you do, I'm happy to approve and merge!

File: docs/release_notes/release_notes-0.14.0.rst [hunk 1]
```diff
@@ -23,6 +23,7 @@ Key Features and Updates
   *
 * Documentation improvements
   * DOCS-#4077: Add release notes template to docs folder (#4078)
+  * DOCS-#4082: Add pdf/epub/htmlzip formats for doc builds (#4083)
```
Comment (prutskov) [inline docs/release_notes/release_notes-0.14.0.rst:26]: @RehanSD I've added a type/issue number/PR name/ PR number as it's described in example above. But I didn't find an example where I should place my github username. Should it be placed in "Contributors" section or not? Example doesn't show where I should put it

Comment (RehanSD) [reply docs/release_notes/release_notes-0.14.0.rst:26]: Yes please! For context, this is the release description you see [here](https://github.com/modin-project/modin/releases/tag/0.13.0), so please add your name to the Contributors section with an `@` so GitHub can also auto link your profile!

Comment (RehanSD) [reply docs/release_notes/release_notes-0.14.0.rst:26]: Type/Issue/Pr Title + Number look perfect! I suspect Devin forgot to add his name to the contributors, which I'll fix when I add my docs fix PR that got merged last night to this template!

**Review by RehanSD — APPROVED** [review]
Looks great to me!

**Review by YarShev — APPROVED** [review]
LGTM

**Pull request merged**